fsm/observer.go

```diff
@@ -46,9 +46,43 @@ func (c *CachedObserver) GetCachedNotifications() []Notification {
 	return c.cachedNotifications.Get()
 }
 
+// WaitForStateOption is an option for the WaitForState function.
+type WaitForStateOption struct {
```

Comment:
Almost, but what you want is:

```
type FSMOption interface {
    apply(*fsmOptions) 
}

type fsmOptions struct {
   initialWait time.Duration
}

type WaitForStateOption struct {
   initialWait time.Duration
}

func WithWaitForStateOption(initialWait time.Duration) FSMOption {
   return &WaitForStateOption{
         initialWait,
   }
}

func (w *WaitForStateOption) apply(o *fsmOptions) {
   o.initialWait = w.initialWait
}

```

And then below in the constructor you can pass in `opts ...FSMOption` and just call `apply()` on them.
This pattern is the same e.g. grpc uses with `DialOption` and we use the same pattern in LND a lot.